crates/turbopack-mdx/src/lib.rs

```diff
@@ -31,12 +34,37 @@ fn modifier() -> StringVc {
     StringVc::cell("mdx".to_string())
 }
 
+/// Subset of mdxjs::Options to allow to inherit turbopack's jsx-related configs
+/// into mdxjs.
+#[turbo_tasks::value(serialization = "auto_for_input")]
+#[derive(PartialOrd, Ord, Hash, Debug, Copy, Clone)]
+pub struct MdxTransformOptions {
+    pub development: bool,
+    pub preserve_jsx: bool,
+    pub jsx_runtime: OptionStringVc,
+    pub jsx_import_source: OptionStringVc,
+    pub filepath: Option<FileSystemPathVc>,
+}
+
+impl Default for MdxTransformOptions {
+    fn default() -> Self {
+        Self {
+            development: true,
+            preserve_jsx: false,
+            jsx_runtime: OptionStringVc::cell(None),
+            jsx_import_source: OptionStringVc::cell(None),
+            filepath: None,
+        }
+    }
+}
+
 #[turbo_tasks::value]
 #[derive(Clone, Copy)]
 pub struct MdxModuleAsset {
     source: AssetVc,
     context: AssetContextVc,
     transforms: EcmascriptInputTransformsVc,
+    options: MdxTransformOptions,
```

Comment:
Instead of having a struct with Vcs, we could have a Vc of values here, which would help the propagation of data without invalidating everything.